.github/actions/paths-filter/action.yml

```diff
@@ -11,14 +11,15 @@ outputs:
         github.event_name == 'push' ||
         steps.filter-common.outputs.github-actions-change == 'true'
       }}
-  java-code-changes:
+  java-code-changes:  # Also runs spotless for Markdown linting
     description: Output whether jobs depending on Java code changes should be run, related checks are based on GitHub events and java files changes
     value: >-
       ${{
         github.event_name == 'push' ||
         steps.filter-common.outputs.github-actions-change == 'true' ||
         steps.filter-common.outputs.java-code-change == 'true' ||
-        steps.filter-common.outputs.maven-change == 'true'
+        steps.filter-common.outputs.maven-change == 'true' ||
+        steps.filter-common.outputs.markdown-change == 'true'
       }}
```

Comment:
❓ I was wondering whether we want to somehow separate this 🤔 to better react on in the ci and only run spotless markdown (if possible) and not everything else 🤔

Reply:
> whether we want to somehow separate this

That would be nice since it makes the whole setup more modular and allows us to only run the necessary functions. For this PR though I want to focus on getting the minimum done and close the loophole, so any refactoring into separate jobs (one for java checks and one for markdown checks) would have to be a follow-up 👍🏼